### ISSUE TITLE
json2dict converts unquoted true and false to numerical boolean

### ISSUE THREAD

**Issue by ramikhaldi** [flightaware/yajl-tcl#38]
The json2dict treats the unquoted string values true and false as boolean and converts them to 1 and 0.
Example:
{
    "valueTrue": true,
    "valueFalse": false
}

will be converted to:

valueTrue 1 valueFalse 0

Is there any configuration to prevent that ?

Thanks in advance!

**Comment by bovine** [comment]
That's an interesting observation. In general, we've tried to make the json2dict function perform identically to the same one provided by the json package, except faster. However, this is indeed a difference in behavior:
```
package require json

::json::json2dict {{
"valueTrue": true,
"valueFalse": false
}}

valueTrue true valueFalse false
```

**Comment by bovine** [comment]
I think changing the replacements in `parse2dict_boolean_callback` might fix this behavior? https://github.com/flightaware/yajl-tcl/blob/master/generic/yajltcl.c#L227

**Comment by ramikhaldi** [comment]
Thanks @bovine for your response, I also have seen that location but I was not sure if there might be other side effects when fixing it. As there are already tests for the library, we can rerun the tests and see if something has been broken. I will recompile the library and check.

**Comment by ramikhaldi** [comment]
@bovine  I have performed a pull request. Let's see the test results in the ci..

**Comment by ramikhaldi** [comment]
Can you please approve the waiting workflows (https://github.com/flightaware/yajl-tcl/pull/39)? Thanks.

**Comment by bovine** [comment]
That PR has been merged and released as v1.8.1
